pandas/core/internals/__init__.py

```diff
@@ -38,3 +36,27 @@
     # this is preserved here for downstream compatibility (GH-33892)
     "create_block_manager_from_blocks",
 ]
+
+
+def __getattr__(name: str):
+    import warnings
+
+    from pandas.util._exceptions import find_stack_level
+
+    if name in ["NumericBlock", "ObjectBlock"]:
+        warnings.warn(
+            f"{name} is deprecated and will be removed in a future version. "
+            "Use NumpyBlock instead.",
```

Comment:
I'm fine with not adding `NumpyBlock` to the semi-public API, but in that case what should I suggest to users as the replacement in the deprecation warning here? Not sure if there's a polite way to say "whatever you're doing you shouldn't be doing" :stuck_out_tongue:.